src/org/lwjgl/opengl/awt/PlatformLinuxGLCanvas.java

```diff
@@ -48,7 +53,14 @@ private long create(int depth, GLData attribs, GLData effective) throws AWTExcep
 			// No framebuffer configurations supported!
 			throw new AWTException("No supported framebuffer configurations found");
 		}
-		long context = glXCreateNewContext(display, fbConfigs.get(0), GLX_RGBA_TYPE, NULL, true);
+
+		copyGLAttribsToEffective(attribs, effective);
```

Comment:
`copyGLAttribsToEffective` does not actually check the effective GL version. This can deviate arbitrarily from the requested OpenGL version, especially then requesting a GL version less than 3.2 or when using the compatibility profile. The Nvidia driver in particular will then just return a context with the highest supported version.
Please have a look at the Windows implementation: https://github.com/LWJGLX/lwjgl3-awt/blob/master/src/org/lwjgl/opengl/awt/PlatformWin32GLCanvas.java#L506-L519
https://github.com/LWJGLX/lwjgl3-awt/blob/master/src/org/lwjgl/opengl/awt/PlatformWin32GLCanvas.java#L670-L713
It goes to some lengths to actually determine the effective version and profile.
Also using an arbitrary "default" version (in your example 3.3) is not sufficient. We _must_ query the _actual_ OpenGL context version.

Reply:
Are you sure? I don't think it can deviate, because from the docs glXCreateContextAttribsARB fails if: 

>       * If attribute GLX_CONTEXT_PROFILE_MASK_ARB has no bits set; has
>         any bits set other than GLX_CONTEXT_CORE_PROFILE_BIT_ARB and
>         GLX_CONTEXT_COMPATIBILITY_PROFILE_BIT_ARB; has more than one of
>         these bits set; or if the implementation does not support the
>         requested profile, then GLXBadProfileARB is generated.

Does nvidia really violate the spec regarding this GLX call?

If you're also concerned that it's possible to request eg 3.2 and then get a 4.0 context due to forward compatibility while effective still says 3.2, I guess that's true. Do you feel strongly it's important to query for that case?

Reply:
This paragraph mentions the case when the GLX_CONTEXT_PROFILE_MASK_ARB is in the attribute list but the value of this attribute is invalid (the three cases mentioned). However, when you leave the attribute altogether, you make a request to obtain a "legacy" <= 2.1 (and given some conditions, also 3.0, 3.1 and 3.2 compatibility) context, which we also must support. See section "3.3.7.1 Legacy Context Creation". In those cases, the context _may_ have any version larger than the requested version.

Reply:
Okay, I will work on adding querying. Regarding the Windows implementation, why call JNI.callPV instead of the wrappers like https://javadoc.lwjgl.org/org/lwjgl/opengl/GL11.html#glGetIntegerv(int,int%5B%5D)?

Also, would you prefer I remove the default 3.3?

Reply:
Oh, because it's avoiding the createCapabilities call. Makes sense.

Reply:
I pushed some changes, would appreciate your feedback if you have a chance!